### PR TITLE
Use Title Caps in Installer

### PR DESCRIPTION
**Pull request by Avinash-Acharya**
Linked to - General - Fix Title Caps , #2000

Signed-off-by : Avinash Acharya <avinashaacharya1@gmail.com>

### CODE REVIEW

**Comment by kingthorin** [comment]
To address the DCO requirement you'll need to sign-off the commit(s):

- https://github.com/zaproxy/zaproxy/blob/main/CONTRIBUTING.md#developer-certificate-of-origin
- https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff

File: zap/src/main/installer/resources/Messages.properties
Comment (thc202) [inline zap/src/main/installer/resources/Messages.properties:15]: Why these changes?

Comment (Avinash-Acharya) [reply zap/src/main/installer/resources/Messages.properties:15]: I was not sure about the instructions and made changes randomly 😔 sorry for that. I was using intillij as IDE and had no idea how to check the gui for these changes.

**Comment by Avinash-Acharya** [comment]
> To address the DCO requirement you'll need to sign-off the commit(s):
> 
> * https://github.com/zaproxy/zaproxy/blob/main/CONTRIBUTING.md#developer-certificate-of-origin
> * https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff

Does mentioning singed-off-by in comments does the job? Sorry for being nuisance, I am new to GitHub 😔.

**Comment by kingthorin** [comment]
No. You can do it via:
```bash
git commit --amend --sign-off
git push --force
```

Assuming you're in the appropriate directory with the appropriate branch checked out.

File: zap/src/main/installer/resources/Messages.properties
Comment (kingthorin) [reply zap/src/main/installer/resources/Messages.properties:15]: Well since they're installer strings you'd have to actually build the installer and check.

I plan to have a look but won't get to it until tomorrow night.

There's developer docs here that'd probably help you: https://www.zaproxy.org/docs/developer/

**Comment by kingthorin** [comment]
Sorry for the delay I didn’t get to this yesterday. Still on my todo list.

**Comment by Avinash-Acharya** [comment]
> Sorry for the delay I didn’t get to this yesterday. Still on my todo list.

I should be the one sorry , you gave me the task and I didn't finished it yet. I shouldn't make excuses but I had tons of Uni work. I will make sure to learn all git commands and correct my change.
One question, should I close this PR and send new one again or is it possible to add "signed-off-by" in an existing PS from terminal.

**Comment by kingthorin** [comment]
You can do it in this PR. The commands below amend the existing commit with your sign-off and then you force push it back to the PR branch. Which should re-trigger the DCO check.

**Comment by Avinash-Acharya** [comment]
> You can do it in this PR. The commands below amend the existing commit with your sign-off and then you force push it back to the PR branch. Which should re-trigger the DCO check.

I need to close this PR (after your approval) because I just found out that I made changes the main branch.
Question :
- Is it possible for me to check the GUI for that changes I made at the installer segment?
- Is there any plugin in the IDE (I use intellij) for GUI ?

**Comment by kingthorin** [comment]
You don't need to close this. Just make sure you use a "feature" branch in the future.

**Comment by kingthorin** [comment]
I've got screenshots of the existing installer where these strings are used. I just need to put some thought into how/where they're used. I'll add them with notes shortly.

**Comment by thc202** [comment]
Also worth checking ZAP itself (for consistency), some of the strings match what's being shown the UI as well.

**Comment by kingthorin** [comment]
image
image

Some of these strings should have periods. But it seems most of those this PR changes are used more as sentences then titles. However, we also need to be careful what ends up being displayed with a colon in the install summary screen (second screenshot).

**Comment by kingthorin** [comment]
Okay here's my take:

- We don't add periods, because most of these are then used with colons.
- "Check for updates" as in "Check for updates on startup", the updates should be lower case because it isn't a name.
- "Add-on" or "Add-ons" should only have a capital if it's leading a string/sentence or is a section/grouping title because it isn't a name.
- Release/Alpha/Beta should have leading caps because they are names.

**Comment by kingthorin** [comment]
Lines 9 and 34, could you change Updates ➡ updates ?

**Comment by kingthorin** [comment]
On the DCO front if you can just let us know that you agree with: https://developercertificate.org/ we can either manually set it to pass or I can update the branch on your behalf once everyone has agreed on the changes.

**Comment by Avinash-Acharya** [comment]
> Okay here's my take:
> 
> * We don't add periods, because most of these are then used with colons.
> * "Check for updates" as in "Check for updates on startup", the updates should be lower case because it isn't a name.
> * "Add-on" or "Add-ons" should only have a capital if it's leading a string/sentence or is a section/grouping title because it isn't a name.
> * Release/Alpha/Beta should have leading caps because they are names.

- colon are not used in first screenshot so should I add periods in them?
- I restored the text to original and added title caps to those you mentioned .

**Comment by kingthorin** [comment]
>   colon are not used in first screenshot so should I add periods in them?

Yes that makes sense, I just noticed that there are separate "summary" resource messages.

>    I restored the text to original and added title caps to those you mentioned .

Thanks.

**Comment by Avinash-Acharya** [comment]
> On the DCO front if you can just let us know that you agree with: https://developercertificate.org/ we can either manually set it to pass or I can update the branch on your behalf once everyone has agreed on the changes.

I agree .

**Comment by kingthorin** [comment]
Setting DCO pass.
image

@zaproxy/team-zaproxy I don't mind flattening this once we're happy with it and applying Avinash's sign-off.

File: zap/src/main/installer/resources/Messages.properties
Comment (kingthorin) [inline zap/src/main/installer/resources/Messages.properties:7]: Just discussed this PR via IRC. I think this is the last thing that needs tweaking.

This is a title for the screen, we feel it seems out of place with a period, so if you could revert this one that'd be great.

image

**Review by kingthorin — APPROVED** [review]
Thanks. I'll flatten this and do the sign-off on your behalf this evening.

**Comment by Avinash-Acharya** [comment]
Thanks ..!

**Review by kingthorin — APPROVED** [review]

**Comment by kingthorin** [comment]
Poke @psiinon @thc202 this good now?

I’ll check that it aligns with the options panel and CFU/Marketplace. (If not I’ll do another PR)

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you both!

**Comment by kingthorin** [comment]
Thanks!

**Comment by Avinash-Acharya** [comment]
Thanks to you all !!

**Pull request merged**
